bankAccount/src/main/kotlin/com/example/bankAccount/adapters/repository/SpringDataAccountRepositoryAdapter.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.bankAccount.adapters.repository
+
+import com.example.bankAccount.adapters.persistence.AccountEntity
+import com.example.bankAccount.adapters.persistence.TransactionEntity
+import com.example.bankAccount.domain.Account
+import com.example.bankAccount.domain.AccountRepository
+import com.example.bankAccount.domain.Transaction
+import org.springframework.stereotype.Repository
+import java.text.SimpleDateFormat
+
+@Repository
+open class SpringDataAccountRepositoryAdapter(private val springDataAccountRepository: SpringDataAccountRepository) :
+    AccountRepository {
+
+    override fun consultAllAccounts(): List<Account> {
+        val accountList = springDataAccountRepository.findAll().map { accountEntity -> accountEntity.toDomain() }
+        return accountList
+    }
+
+    override fun consultAccount(iban: String): Account? {
+        val account = springDataAccountRepository.findByIban(iban)?.toDomain()
+        return account
+    }
+
+    override fun openAccount(account: Account): Account {
+        val accountEntity = account.toEntity()
+        springDataAccountRepository.save(accountEntity)
+        return account
+    }
+
+    override fun saveAccount(account: Account) {
+        val accountEntity = account.toEntity()
+        springDataAccountRepository.save(accountEntity)
+    }
+
+    override fun closeAccount(iban: String) {
+        springDataAccountRepository.deleteById(iban)
+    }
+}
+
+// Mappage de AccountEntity à Account
+private fun AccountEntity.toDomain(): Account {
+    return Account(
+        iban = this.iban,
+        firstName = this.firstName,
+        lastName = this.lastName,
+        balance = this.balance,
+        transactions = this.transactions.map { transactionEntity -> transactionEntity.toDomain() }.toMutableList()
```

Comment:
En Kotlin, si un lambda n'a qu'un seul paramètre, il est implicitement nommé "it", et c'est très pratique:

```suggestion
        transactions = this.transactions.map { it.toDomain() }.toMutableList()
```